src/domains/ecommerce-common/03_storage.tf

```
@@ -378,138 +283,23 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_transactions_c
       OpCountForQueue("PutMessage", queueKey)
       | join OpCountForQueue("DeleteMessage", queueKey) on count_
       | project name = queueKey, Count = count_ - count_1
+      | where toint(Count) > ${each.value.threshold}
     };
     MessageRateForQueue("%s")
```

I would move the thresholding outside of the rate computation, wdyt?

```suggestion
      | where toint(Count)
    };
    MessageRateForQueue("%s") > ${each.value.threshold}
```

---

I'm not sure it works.
`MessageRateForQueue("%s")` doesn't return an int but it returns a table like


```
| COUNT |
|---------|
|       0      |
----------
```

I don't think you can compare a table with an int value.

The `where` conditions returns an empty table. 
The result size is matched against trigger threshold value to decide if fire an alert or not.